rules/engine.go

```diff
@@ -118,6 +121,10 @@ func (e *v3Engine) SetKVWrapper(kvWrapper WrapKV) {
 	e.kvWrapper = kvWrapper
 }
 
+func (e *v3Engine) SetWatchWrapper(kvWatcher WrapWatcher) {
```

Comment:
I think it should be wrapping `SetWatcherWrapper` since we're wrapping the `Watcher` not the `Watch` instance